test/core/renderControllerTests.ts

```diff
@@ -1,8 +1,80 @@
 ///<reference path="../testReference.ts" />
 
 describe("RenderController", () => {
-  // HACKHACK: #2083
-  it.skip("Components whose render() is triggered by another Component's render() will be drawn", () => {
+  let SVG_WIDTH = 400;
+  let SVG_HEIGHT = 300;
+
+  describe("configuring the render policy", () => {
+    let oldWarn: (msg: string) => void;
+    let storedWarningMsg: string;
+
+    before(() => {
+      oldWarn = Plottable.Utils.Window.warn;
+      Plottable.Utils.Window.warn = (warningMsg: string) => {
```

Comment:
This is getting scarier and scarier. On top of the above, can we do the following. 

In the `globalInitialization.ts`, in the `before()` make a copy of all the methods which we hack like this (warn, setTimeout, etc) and in the `after()` assert that we restored all the methods to their original state.

Regarding the other question, what exactly is the warning we are catching here?